tools/CopyrightDateTool.scala

```diff
@@ -0,0 +1,171 @@
+    val lines = try source.mkString finally source.close()
+    lines
+  }
+
+  def getGitCreationYear(pathToFile: String): String = {
+    val buf_dates = (s"git log   --format=%ci $pathToFile" !!)
+    val dates = buf_dates.split("\\n").distinct
+    val year_column = 0
+    val res = dates.last.split("-")(year_column)
+    res
+  }
+
+  def getGitLastUpdateYear(pathToFile: String): String = {
+    val buf_dates = (s"git log -3  --format=%cd $pathToFile" !!)
+    val buf_messages = (s"git log -3  $pathToFile" !!)
+    val first_empty_elem = 1
+    val messages = buf_messages.split("commit").distinct.drop(first_empty_elem)
+    val dates = buf_dates.split("\\n").distinct
+    val gits_info = (messages, dates).zipped.map{ case(a,b) => GitInfo(a,b)}
+    val res = gits_info.filter(!_.message.contains("[Sync copyright dates]"))
+    val year_column = 4
+    val first = 0
+    res(first).date.split(" ")(year_column)
+  }
+
+  def getYear(pathToFile: String, lower_bound: Int, upper_bound: Int): String = {
+    val file = readFile(pathToFile)
+    val third_line = 2
+    val start_index = file.split("\\n")(third_line).indexOf("(c) 20")
+    val year = file.split("\\n")(third_line)
+                   .substring(start_index + lower_bound,
+                              start_index + upper_bound)
+    year
+  }
+
+  def getFileCreationYear(pathToFile: String): String = {
+    val lb = "(c) ".length()
+    val ub = "(c) 20XX".length()
+    val fileCreationYear = getYear(pathToFile, lb, ub)
+    fileCreationYear
+  }
+
+  def getFileLastUpdateYear(pathToFile: String): String = {
+    val lb = "(c) 20XX-".length()
+    val ub = "(c) 20XX-XXXX".length()
+    val fileLastUpdateYear = getYear(pathToFile, lb, ub)
+    fileLastUpdateYear
+  }
+
+  def analyzeDates(): Array[FileDates] = {
+    val files_array = readFilesList()
+    var data = Array[FileDates]()
+    for(file_path <- files_array) {
+      val fileLastUpdateYear = getFileLastUpdateYear(file_path)
+      val gitLastUpdateYear = getGitLastUpdateYear(file_path)
+      val fileCreationYear = getFileCreationYear(file_path)
+      val gitCreationYear = getGitCreationYear(file_path)
+      data = data :+ FileDates(file_path,
+                      fileLastUpdateYear,
+                      gitLastUpdateYear,
+                      fileCreationYear,
+                      gitCreationYear)
+    
+    }
+    data
+  }
+
+  def fixWrongDates(): Boolean = {
+    var check = false
+    val files = analyzeDates()
+    files.foreach { file =>
+      val file_file = readFile(file.pathToFile)
+      var new_file: Option[String] = None
+      if (file.fileLastUpdateYear.forall(_.isDigit)) {
+        if (file.isUpdateDateWrong()) {
+          new_file =Some(file_file.
+                       replace(
+                        s"${file.fileCreationYear}-${file.fileLastUpdateYear}",
+                        s"${file.fileCreationYear}-${file.gitLastUpdateYear}"
```

Comment:
This replacement would be more flexible with the regex match, as well as safer. For example, you would replace the string constants in the scaladoc sources by doing global replace on the file contents. And you could replace "(c) 2009" with "Copyright 2009-2016" or however it ought to be normalized.